src/containers/Books/Books.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { useHistory } from "react-router-dom";
+import BookCard from '../../components/BookCard/BookCard';
+import PaginationButton from '../../components/PaginationButton/PaginationButton';
+import SearchPanel from '../../components/SearchPanel/SearchPanel';
+import Loader from '../../components/Loader/Loader';
+import { getBooks } from '../../API/fetch';
+
+function Books() {
+  const [data, setData] = useState([]);
+  const [resultsValue, setResultsValue] = useState(0);
+  const [bookTitle, setBookTitle] = useState('');
+  const [sort, setSort] = useState('');
+  const [category, setCategory] = useState('');
+  const [loaderVisibility, setLoaderVisibility] = useState(true);
+  const history = useHistory();
+
+  const onFinish = (bookTitle, sort = 'relevance', category = 'all') => {
+    if (bookTitle) {
+      setLoaderVisibility(false);
+      setBookTitle(bookTitle);
+      setSort(sort);
+      setCategory(category);
+
+      getBooks(bookTitle, sort, category)
+        .then(data => {
+            setLoaderVisibility(true);
+            if (data.totalItems) {
+                setData(data.items);
+                setResultsValue(data.totalItems);
+            } else {
+                setData([]);
+                setResultsValue(data.totalItems);
+            }
+      });
+    }
+  };
+
+  const loadMore = (bookTitle, sort, category, startIndex, maxResults) => {
+    setLoaderVisibility(false);
+
+    getBooks(bookTitle, sort, category, startIndex, maxResults)
+        .then(json => {
+            setLoaderVisibility(true);
+            setData(data.concat(json.items));
+            setResultsValue(json.totalItems);
+      });
+    if (maxResults < 30) setBookTitle('');
+  };
+
+  const onBookClick = (id) => {
+    history.push(`/card/${id}`);
```

Comment:
there is a const with urls, you should use it instead of string literals all the time

Reply:
Forgot to do it